src/sbvr-api/resource-access.ts

```diff
@@ -0,0 +1,153 @@
+import * as express from 'express'
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+import TypedError = require('typed-error')
+import { AnyObject } from 'pinejs-client/core';
+import PinejsClient = require('pinejs-client');
+
+class BadInputError extends TypedError {}
+class UnauthorizedError extends TypedError {}
+
```

Comment:
Similar errors are generated and exported in sbvr-utils and uri-parser, maybe have a look to see if there is any overlap with those

Reply:
I would rather stay with these errors. uri-parser exports a `BadRequestError` but it is still in coffee script and I did not find a similar `UnauthorizedError`.

Reply:
I'd strongly prefer to just use the same `BadRequestError` in both places, right now any code that might hit either of these has to separately handle both of these errors which will almost certainly lead to cases where only one is handled and the other slips through, despite being essentially the same thing